### PR TITLE
#45949 & #45950 Divide workflows page into three sections & Update payment section with bank icon

### PR DESCRIPTION
**Pull request by Guccio163**
### Fixed Issues

$ https://github.com/Expensify/App/issues/45949
$ https://github.com/Expensify/App/issues/45950
PROPOSAL:

### Tests

Open workflows page (turn it on in 'more features' page).

Sections should look like this now:
Screenshot 2024-07-26 at 13 24 15

Verify that each section and its children are clickable

Verify that each section looks correct (with or without a connected bank account):
Delay submissions:
Screenshot 2024-07-26 at 13 20 30
Add approvals:
Screenshot 2024-07-26 at 13 22 45
Payments (with added account):
Screenshot 2024-07-26 at 13 22 58
Payments (without added account):
Screenshot 2024-07-26 at 13 23 30

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

N/A

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/a4bc0795-1e48-4a83-adc0-5cfcf17b5abd

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/2f883509-5f2f-4e28-98f3-5001a27b7806

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/77256c03-d353-4fba-9d4e-b8162a38a92d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/e78093e6-eef8-4858-ad50-55781e5058d7

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/9532fe9b-a835-4421-9141-86446d455f60

https://github.com/user-attachments/assets/6532adce-5034-400c-9cc1-420e18f2aa0f

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/6a7d9903-0703-4e72-9442-bc9d00b140c0

->

</details>

### CODE REVIEW

**Comment by Guccio163** [comment]
Right now the subMenuItems are rendered with 2px padding left, ideally in the doc they are aligned to title, but the background that lights up on hover has auto width and would look worse with perfectly aligned text;
Doc:
Screenshot 2024-07-23 at 15 44 16
Aligned:
Screenshot 2024-07-23 at 15 46 00
Right now:
Screenshot 2024-07-23 at 15 46 36
</br>
This is not quite the issue for this ticket, because many subMenuItems will be changed too, but I wanted to make it right at once, what do you think?

File: src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ function WorkspaceWorkflowsPage({policy, betas, route}: WorkspaceWorkflowsPagePr
 
     const policyApproverEmail = policy?.approver;
     const policyApproverName = useMemo(() => PersonalDetailsUtils.getPersonalDetailByEmail(policyApproverEmail ?? '')?.displayName ?? policyApproverEmail, [policyApproverEmail]);
-    const containerStyle = useMemo(() => [styles.ph8, styles.mhn8, styles.ml11, styles.pv3, styles.pr0, styles.pl4, styles.mr0, styles.widthAuto, styles.mt4], [styles]);
+    const containerStyle = useMemo(() => [styles.ph8, styles.mhn5, styles.ml0, styles.pv3, styles.pr0, styles.pl2, styles.mr0, styles.widthAuto, styles.mt4, styles.pb0], [styles]);
```
Comment (blazejkustra) [inline src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx:54]: You can merge styles like `ml0` and `mr0` into `mh0`, same for the paddding:

This should be sufficient (try it before commiting 😅):
```ts
[styles.pt3, styles.pr0, styles.pl2, styles.pb0, styles.mh0, styles.mt4, styles.widthAuto]
```

File: src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx [hunk 2]
```diff
@@ -252,15 +250,14 @@ function WorkspaceWorkflowsPage({policy, betas, route}: WorkspaceWorkflowsPagePr
         theme,
     ]);
 
-    const renderOptionItem = (item: ToggleSettingOptionRowProps, index: number) => (
-        <View
-            style={styles.mt7}
-            key={`toggleSettingOptionItem-${index}`}
+    const renderOptionItem = (item: ToggleSettingOptionRowProps) => (
+        <Section
```
Comment (blazejkustra) [inline src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx:254]: Let's not remove key from the component, I think it is necessary 🤔

**Review by blazejkustra — CHANGES_REQUESTED** [review]

**Review by blazejkustra — APPROVED** [review]
Code looks clean, no remarks from my side :smile: Great work!

- Let's resolve the conflicts first before recording videos
- Could you update the title to the same title as in the issue?
- Add some test steps so that C+ and QA team can test the changes (it should be rather straight forward)

> This issue is clearly UI-based, after visiting https://dev.new.expensify.com:8082/settings/workspaces/:policyID/workflows it should look like in the docs.

It's better not to link the page (dev version is linked and QA can't access it). Also QAs don't have access to the doc I suppose. Usually I treat this section as a step guide how to test your changes:
- Open workflows page (turn it on in 'more features' page).
- Sections should look like this now:
image

- Verify that each section is clickable
- Verify that each section looks correct (with or without a connected bank account):
image
image
image

File: src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx (hunk 1 above)
Comment (Guccio163) [reply src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx]: Container style is no longer needed, so it solve itself haha :)

**Comment by Guccio163** [comment]
This PR is ready for review, I am aware that there is still need to:
- change some texts that are currently confirmed by Spanish contributors
- resolve esLint error with using empty `Fragment` in code
I will address these issues in the next commit. Also, please take a quick look @shawnborton if the UI is OK 👍

**Comment by blazejkustra** [comment]
@jayeshmangwani Don't worry about pod diff, Wiktor is going to remove it on Monday, cheers!

**Comment by jayeshmangwani** [comment]
> Wiktor is going to remove it on Monday, cheers!

Thanks for the heads up! Enjoy the weekend 😃  !

**Comment by Guccio163** [comment]
I had a moment to take a look at it, just fixed lint check and pods, enjoy your weekend too ✌️

**Comment by shawnborton** [comment]
Can you make sure the top part matches other cards in the workspace editor from a font size perspective?
CleanShot 2024-07-29 at 08 54 52@2x

It looks like the text below the title is too small. Let's also make sure we're using the same gap between, etc.

**Comment by shawnborton** [comment]
Looking good to me!

**Comment by jayeshmangwani** [comment]
There's an issue with the toggle functionality, it's flickering. Initially, the delay submission is enabled.

Sequence of actions:

1. Toggle disabled
2. Toggle enabled

The output shows the toggle switch flickering between states:

1. Disabled
2. Enabled
3. Disabled
4. Enabled

https://github.com/user-attachments/assets/d0b60195-4317-47b5-82cf-d14e44d8c5a5

**Comment by jayeshmangwani** [comment]
> There's an issue with the toggle functionality, it's flickering. 

I've checked with staging, and it seems this issue is occurring there as well. 
It's not coming from this PR, but I will report it, and we will need to handle it in a separate issue.

**Comment by Guccio163** [comment]
That's true, it occurred like that to me too, I think I even caught it in one of my recordings; I guess it is due to work over this section on backend, because there is also a problem with displaying card after switching off one time in `Payments`:

Screenshot 2024-07-30 at 09 57 49

https://github.com/user-attachments/assets/66777c9a-bcf6-4d49-a0fa-0f203fff5649

If it suits you, you can report it with the flickering too :)

**Comment by jayeshmangwani** [comment]
> If it suits you, you can report it with the flickering too :)

Yes sure 👍

**Comment by jayeshmangwani** [comment]
@Guccio163 @blazejkustra In this PR, there is no "Add Approval Workflow" button in the Add Approvals section, and the UI for the Expense form and Approvals is different. Before approving the PR, I just wanted to make sure whether we are going to handle this in another PR or if we need to handle it in this PR.
<details>
  <summary>UI in the Design Doc</summary>
  
Screenshot 2024-07-30 at 2 47 52 PM
</details>
<details>
  <summary>UI in this PR</summary>
  
Screenshot 2024-07-30 at 2 48 51 PM

</details>

**Comment by Guccio163** [comment]
@jayeshmangwani We are going to handle it in [this issue](https://github.com/Expensify/App/issues/45951), this PR is just for splitting the sections, without `Approvals` or connected sidebars/forms

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/d2353eca-f63f-4997-ba58-e99235ada775




</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/db0e81c6-aef1-4d8c-a8e4-ceac67a6ed1d




</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/user-attachments/assets/c0dc8068-9a92-4ca2-9afc-89697c3a008e




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/ca045ddc-d7f2-4ad2-8bd1-843a6ebf8a37




</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/ad463371-65ea-4035-a36d-177ffceca821




</details>

**Comment by jayeshmangwani** [comment]
The changes look good, and I assume that we are going to change the `makeOrTrackPaymentsTitle` in another PR.

https://github.com/Expensify/App/blob/88045884386cd1b69757ad655868772548f881b9/src/languages/en.ts#L1250

**Comment by Guccio163** [comment]
I'm currently waiting for this and 4 other translations, I'd like to change it in this PR :)

**Comment by jayeshmangwani** [comment]
Great, let's wait for the translations; otherwise, it looks good.

**Comment by Guccio163** [comment]
@jayeshmangwani  I added translations, they are all approved so it's ready for merge 👌

**Review by jayeshmangwani — APPROVED** [review]
LGTM 🚀

**Review by tgolen — APPROVED** [review]

**Pull request merged**